api/v1alpha1/tls_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// TLSConfiguration describes the TLS configuration for a party in a TLS
+// session.
+//
+// References
+// - nginx: https://nginx.org/en/docs/http/configuring_https_servers.html
+// - envoy: https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/auth/cert.proto
+// - haproxy: https://www.haproxy.com/documentation/aloha/9-5/traffic-management/lb-layer7/tls/
+// - gcp: https://cloud.google.com/load-balancing/docs/use-ssl-policies#creating_an_ssl_policy_with_a_custom_profile
+// - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
+// - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
+type TLSConfiguration struct {
```

Comment:
As I mentioned in on a related issue, for the intial API, most these things will be placed in implementation-specific extensions. 

We unfortunately need to consider a subset for the core and extended APIs.

Reply:
This doesn't change the items that can be configured much. The main thing here is that I break out an explicit (but currently un-specced) object for TLS peer validation. I think that's a useful approach that will help us manage the different concerns around TLS config.

Reply:
For usability I would like to see a more simplified names, i.e. `TLSConfig` or `TLSOptions`.